File: resharper/test/data/References/MemberDataFromDerivedClass.vb
Imports System
Imports System.Collections.Generic
Imports Xunit

Public Class MyTest
    Inherits MemberDataBase

    <Theory>
    <MemberData("DataEnumerator")>
    Public Sub Test1(ByVal value As Int32)
        Assert.Equal(42, value)
    End Sub
End Class

Public Class MemberDataBase
    Public Shared ReadOnly Iterator Property DataEnumerator() As IEnumerable(Of Object())
        Get
            Yield New Object() { 42 }
        End Get
    End Property
End Class

File: resharper/test/data/References/MemberDataFromDerivedClass.cs
using System.Collections.Generic;
using Xunit;

namespace Foo
{
    public class MyTest : MemberDataBase
    {
        [Theory]
        [MemberData("DataEnumerator")]
        public void DataComesFromProperty(int value)
        {
            Assert.Equal(42, value);
        }
    }

    public class MemberDataBase
    {
        public static IEnumerable<object[]> DataEnumerator
        {
            get { yield return new object[] { 42 }; }
        }
    }
}
